File: leetcode.com/0206_reverse_linked_list/solution.py
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        values = []
        while head:
            values.append(head.val)
            head = head.next

        head = ListNode()
        node = head

        for val in reversed(values):
            node.next = ListNode(val)
            node = node.next

        return head.next

File: leetcode.com/0206_reverse_linked_list/solution.go
func reverseList(head *ListNode) *ListNode {
    if head == nil {
        return head
    }

    var values []int

    for head != nil {
        values = append(values, head.Val)
        head = head.Next
    }

    node := &ListNode{}
    head = node

    for i := len(values) - 1; i >= 0 ; i-- {
        node.Val = values[i]

        if i != 0 {
            node.Next = &ListNode{}
            node = node.Next
        }
    }

    return head
}
